### PR TITLE
Ensure project creation form does not allow invalid dates,

### PR DESCRIPTION
**Pull request by kluchrj**
add 1 day to throughput calculation.

### CODE REVIEW

File: timepiece/crm/forms.py [hunk 1]
```diff
@@ -135,6 +135,30 @@ def clean(self):
...
 
+            target_internal_completion_date = cleaned_data.get('target_internal_completion_date')
+            required_completion_date = cleaned_data.get('required_completion_date')
+            target_open_date = cleaned_data.get('target_open_date')
+            start_date = cleaned_data.get('start_date')
+            turn_in_date = cleaned_data.get('turn_in_date')
+
+            if start_date:
+                if target_internal_completion_date and target_internal_completion_date < start_date:
+                    self._errors['target_internal_completion_date'] = self.error_class(
```
Comment (dannybrowne86) [inline timepiece/crm/forms.py]: It is not good practice to directly interact with private variables (that's what the _ indicates).  I suggest you update this code to use this function: https://docs.djangoproject.com/en/1.8/ref/forms/api/#django.forms.Form.add_error

File: timepiece/reports/views.py [hunk 2]
```diff
@@ -2241,12 +2241,11 @@ def get_context_data(self, **kwargs):
                 turn_in_date = milestones.get(name='Turn-In')
                 required_completion_date = milestones.get(
                     name='Required Completion')
-                required_time = (required_completion_date.due_date -
-                    start_date.due_date) + datetime.timedelta(days=1)
-                spent_time =  (turn_in_date.due_date -
-                    start_date.due_date) + datetime.timedelta(days=1)
+                required_time = (required_completion_date.due_date - 
+                                 start_date.due_date)
+                spent_time = (turn_in_date.due_date - start_date.due_date)
```
Comment (dannybrowne86) [inline timepiece/reports/views.py:2246]: Are we sure we do not want to add a `1.0` here?  Throughput is zero if we don't...
Maybe what we do is if `turn_in_date.due_date == start_date.due_date` we look at time charged against the project or `ActivityGoals` against the project?

File: timepiece/crm/forms.py (hunk 1 above)
Comment (kluchrj) [inline timepiece/crm/forms.py]: Done.

File: timepiece/reports/views.py (hunk 2 above)
Comment (kluchrj) [inline timepiece/reports/views.py:2246]: Craig has indicated that a throughput of 0.0 is fine for projects that start and complete on the same day, and also add a day to the required_time if the completion date and start date are the same.

**Pull request merged**